src/System Application/App/AppSource Gallery/src/AppSourceProductList.Page.al

```diff
@@ -0,0 +1,208 @@
+                }
+                field(PublisherID; Rec.PublisherID)
+                {
+                    Visible = false;
+                    ToolTip = 'Specifies the Publisher ID';
+                }
+
+                field(PublisherDisplayName; Rec.PublisherDisplayName)
+                {
+                    ToolTip = 'Specifies the Publisher Display Name';
+                }
+                field(Installed; CurrentRecordCanBeUninstalled)
+                {
+                    Caption = 'Installed';
+                    ToolTip = 'Specifies if this app is installed';
+                }
+                field(RatingAverage; Rec.RatingAverage)
+                {
+                    ToolTip = 'Specifies the Rating Average';
+                }
+                field(Popularity; Rec.Popularity)
+                {
+                    ToolTip = 'Specifies the Popularity index';
+                }
+                field(RatingCount; Rec.RatingCount)
+                {
+                    ToolTip = 'Specifies the number of times this has been rated';
+                    Visible = false;
+                }
+                field(LastModifiedDateTime; Rec.LastModifiedDateTime)
+                {
+                    ToolTip = 'Specifies the date and time this was last modified';
+                }
+                field(AppID; Rec.AppID)
+                {
+                    ToolTip = 'Specifies theappID';
+                    Visible = false;
+                }
+                field(PublisherType; Rec.PublisherType)
+                {
+                    ToolTip = 'Specifies the Publisher Type';
+                }
+            }
+        }
+    }
+
+    actions
+    {
+        area(Promoted)
+        {
+            actionref(AppSource_Promoted; OpenAppSource) { }
+            actionref(Open_Promoted; OpenInAppSource) { }
+            actionref(Refresh_Promoted; UpdateProducts) { }
+            actionref(ShowSettings_Promoted; ShowSettings) { }
+        }
+
+        area(Processing)
+        {
+            action(OpenAppSource)
+            {
+                Caption = 'View AppSource';
+                Scope = Repeater;
+                Image = OpenWorksheet;
+                ToolTip = 'View all apps in AppSource';
+
+                trigger OnAction()
+                begin
+                    AppSourceProductManager.OpenAppSource();
+                end;
+            }
+
+            action(OpenInAppSource)
+            {
+                Caption = 'View in AppSource';
+                Scope = Repeater;
+                Image = Open;
+                ToolTip = 'View selected app in AppSource';
+
+                trigger OnAction()
+                var
+                    SelectedRec: Record "AppSource Product";
+                begin
+                    CurrPage.SetSelectionFilter(SelectedRec);
+                    SelectedRec.Next();
+
+                    if (SelectedRec.Count() = 1) then
```

Comment:
I would instead just open the record the user is currently on, that avoids the error as well.

Reply:
Next PR